test/e2e/v1beta1/scripts/gh-actions/verify.py

```diff
@@ -0,0 +1,143 @@
+    ):
+        raise Exception(
+            "All Trials must be Completed. Max Trial count: {}, Experiment status: {}".format(
+                max_trial_count, experiment.status
+            )
+        )
+
+    # If Experiment is Succeeded because of Goal reached, the metrics must be correct.
+    if succeeded_reason == "ExperimentGoalReached" and (
+        (
+            experiment.spec.objective.type == "minimize"
+            and float(best_objective_metric.min) > float(experiment.spec.objective.goal)
+        )
+        or (
+            experiment.spec.objective.type == "maximize"
+            and float(best_objective_metric.max) < float(experiment.spec.objective.goal)
+        )
+    ):
+        raise Exception(
+            "Experiment goal is reached, but metrics are incorrect. "
+            f"Experiment objective: {experiment.spec.objective}. "
+            f"Experiment best objective metric: {best_objective_metric}"
+        )
+
+    # Verify Suggestion's resources. Suggestion name = Experiment name.
+    suggestion = katib_client.get_suggestion(exp_name, exp_namespace)
+
+    # For the Never or FromVolume resume policies Suggestion must be Succeeded.
+    # For the LongRunning resume policy Suggestion must be always Running.
+    for c in suggestion.status.conditions:
+        if (
+            c.type == constants.EXPERIMENT_CONDITION_SUCCEEDED
+            and c.status == constants.CONDITION_STATUS_TRUE
+            and experiment.spec.resume_policy == "LongRunning"
+        ):
+            raise Exception(
+                f"Suggestion is Succeeded while Resume Policy is {experiment.spec.resume_policy}."
+                f"Suggestion conditions: {suggestion.status.conditions}"
+            )
+        elif (
+            c.type == constants.EXPERIMENT_CONDITION_RUNNING
+            and c.status == constants.CONDITION_STATUS_TRUE
+            and experiment.spec.resume_policy != "LongRunning"
+        ):
+            raise Exception(
+                f"Suggestion is Running while Resume Policy is {experiment.spec.resume_policy}."
+                f"Suggestion conditions: {suggestion.status.conditions}"
+            )
+
+    # For Never and FromVolume resume policies verify Suggestion's resources.
+    if (
+        experiment.spec.resume_policy == "Never"
+        or experiment.spec.resume_policy == "FromVolume"
+    ):
+        resource_name = exp_name + "-" + experiment.spec.algorithm.algorithm_name
+
+        # Suggestion's Service and Deployment should be deleted.
+        for i in range(10):
+            try:
+                client.AppsV1Api().read_namespaced_deployment(
+                    resource_name, exp_namespace
+                )
+            except client.ApiException as e:
+                if e.status == 404:
+                    break
+                else:
+                    raise e
+            # Deployment deletion might take some time.
+            time.sleep(1)
```

Comment:
```suggestion
```

I was not sure the reason why we need to add this additional sleep since it seems that we already retry 10 times.
Could you explain the reason why we need additional sleep separately from retry logic?

Reply:
I didn't add any code to this function. I just copy it from the original file.

Reply:
@tenzen-y Do you think that we should remove it?

Reply:
If it is possible, it would be better.